### ISSUE TITLE
Small fix for ficbook.net

### ISSUE THREAD

**Issue by AndyScull** [JimmXinu/FanFicFare#434]
Hi, recently the site's web programmers stopped adding newline br/ tag in html text. They think usual text newline is enough, just because modern browsers can interpret it correctly. But any normal external viewer using strict html standards will just show a big wall of text without linefeeds.
I already reported this bug to them but still no fix.
Had to manually edit adapter_ficbooknet.py and change last lines to:
```
        text = self.utf8FromSoup(url,chapter)
        text = re.sub("(?<!<br>|br/>|r />)[\r\n]{1,2}", "<br/>\n", text, 0, re.I|re.S)
        return text

```
Not sure if you want to add it to official plugin, I'd rather wait until they fix it, seeing no issues about the site here. Hope it helps if anyone has same problem with that site

**Comment by JimmXinu** [comment]
What's going on there is that they are using CSS to tell the browser to interpret the newlines.  Rather than changing the code, it looks cleaner to me to add that CSS to the ebook by including this in `defaults.ini`:

```
[www.ficbook.net]
...
## ficbook.net isn't putting html line breaks in anymore and using
## this CSS instead.
add_to_output_css:
 .part_text {
    word-break: break-word;
    white-space: pre-wrap;
 }
```
This works in the viewers I've tried.  You can try it out in `personal.ini` as well.  Please let me know if it works as expected or not for you.

**Comment by AndyScull** [comment]
Thanks, it was nice to learn something new
I doubt my reader will support it - it's alreader2, last version was released around 2006 and I doubt it's even available for download now. It's very lightweight and reads all formats, the main reason I'm using it is because it stores bookmarks in separate file in same folder as e-book. This makes it very easy to sync my reading progress with just any cloud file storage service.
But I just noticed there's Calibre v4 now, gonna try it's new viewer today, maybe I'll find it usable
